src/System.Management.Automation/engine/ComInterop/ComEventsSink.Extended.cs

```diff
@@ -18,10 +18,7 @@ private void Initialize(object rcw, Guid iid)
         public void AddHandler(int dispid, object func)
         {
             ComEventsMethod method = FindMethod(dispid);
-            if (method == null)
-            {
-                method = AddMethod(dispid);
-            }
+            method ??= AddMethod(dispid);
```

Comment:
@daxian-dbw I see .Net Runtime team already did such changes in the code
https://github.com/dotnet/runtime/blob/e55c908229e36f99a52745d4ee85316a0e8bb6a2/src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/ComEventsSink.Extended.cs#L23

Should we follow original code or we can accept the PR?

Reply:
Didn't know `ComInterop` was taken from Runtime repo and I missed the warning to not modify it, sorry. If we do go along with changing it, I'd vote to take the version from Runtime repo.

Reply:
@Molkree you could copy latest code from Runtime repo for every change _from the PR_ (for the ComInterop) to keep consistency but please wait @daxian-dbw conclusion.

Reply:
I have looked through that PR and it has a couple of places that I have missed. Or should I say Code Analysis has missed 🤔
This might mean there are more missed cases elsewhere which is not good...

Reply:
There was a fix in Runtime repo a few days ago for the analyzer. As only we get it we can fix more issue in PowerShell repo.

Reply:
@iSazonov, when can we expect this fix to be in PowerShell repo? Is it part of .NET Preview 7? I would appreciate the link or any hint on how to find it.

Reply:
@daxian-dbw, if you greenlight even more changes I'd like to bring three more cases from that .NET Runtime PR that were missed by the Analyzer when I crafted this PR.

https://github.com/dotnet/runtime/pull/71361/files#diff-da4c153060d0622e6e18f79a749ebe1b19bf87807a7b0db91e8c4a4220b95ab2L143-R133

https://github.com/dotnet/runtime/pull/71361/files#diff-38d952862014738929f159fe6414fe9e511026ec67e1aba50bb2c9df6bfbee99L22-R22

https://github.com/dotnet/runtime/pull/71361/files#diff-a99f7fa7bdf723c83e65d7e896f006c55db020391658f1636267429911200ab2L107-R109

Reply:
@iSazonov Thanks for spotting that!
@Molkree Please don't bring any .NET changes in this PR. Let's keep this PR only style changes for the existing code. For new changes in dotnet/runtime related to `ComInterop`, please use a separate PR to port the changes.

Reply:
> @iSazonov, when can we expect this fix to be in PowerShell repo? Is it part of .NET 7 Preview 7? I would appreciate the link or any hint on how to find it.

I don't know. I guess it will be in Preview8 - too little time before Preview7.

Reply:
@daxian-dbw


> Let's keep this PR only style changes for the existing code.

Just want to clarify that the additional changes I was talking about are also null-coalescing assignment style changes. No functional changes and no new code.

![image](https://user-images.githubusercontent.com/12860284/181002514-24cb4418-a3d8-4005-941f-0cb9c192c99b.png)
![image](https://user-images.githubusercontent.com/12860284/181002634-9aa22191-866e-47a5-83ff-e912b19dbd55.png)
![image](https://user-images.githubusercontent.com/12860284/181002722-2681f2e8-4dcf-41a9-a723-471e64c5abef.png)

Reply:
Then that's fine I think :) Thanks for double checking

Reply:
@daxian-dbw, pushed these